taipy/gui/gui.py

```diff
@@ -1119,6 +1118,14 @@ def __update_state_context(self, payload: dict):
     def __request_data_update(self, var_name: str, payload: t.Any) -> None:
         # Use custom attrgetter function to allow value binding for _MapDict
         newvalue = _getscopeattr_drill(self, var_name)
+        resource_handler = get_current_resource_handler()
+        custom_page_filtered_types = resource_handler.data_layer_supported_types if resource_handler else ()
+        if (
+            not isinstance(newvalue, _TaipyData)
```

Comment:
not necessary if you remove _TaipyData from the list

Reply:
Still want to keep taipydata in the list so maybe next time :D